src/testSetup/groovy/com/bmuschko/gradle/docker/TestPrecondition.groovy

```diff
@@ -1,34 +1,40 @@
 package com.bmuschko.gradle.docker
 
 final class TestPrecondition {
-    public static final List<String> ALLOWED_PING_PROTOCOLS = ['http', 'https']
+    public static final List<String> ALLOWED_PING_PROTOCOLS = ['tcp', 'http', 'https', 'unix']
     public static final boolean DOCKER_SERVER_INFO_URL_REACHABLE = isDockerServerInfoUrlReachable()
     public static final boolean DOCKER_PRIVATE_REGISTRY_REACHABLE = isPrivateDockerRegistryReachable()
     public static final boolean DOCKERHUB_CREDENTIALS_AVAILABLE = hasDockerHubCredentials()
 
     private TestPrecondition() {}
 
     private static boolean isDockerServerInfoUrlReachable() {
-        isUrlReachable(new URL("${TestConfiguration.dockerHost}/info"))
+        isUrlReachable(new URI("${TestConfiguration.dockerHost}"), "_ping")
     }
 
     private static boolean isPrivateDockerRegistryReachable() {
-        isUrlReachable(new URL(TestConfiguration.dockerPrivateRegistryUrl))
+        isUrlReachable(new URI(TestConfiguration.dockerPrivateRegistryUrl), null)
     }
 
-    private static boolean isUrlReachable(URL url) {
-        if(!ALLOWED_PING_PROTOCOLS.contains(url.protocol)) {
-            throw new IllegalArgumentException("Unsupported URL protocol '$url.protocol'")
+    private static boolean isUrlReachable(URI url, String part) {
```

Comment:
If we change parameter type from `URL` to `URI` IMO we should also change method name to `isUriReachable` and parameter to `uri`.